commands/command_status.go

```diff
@@ -30,41 +30,99 @@ func statusCommand(cmd *cobra.Command, args []string) {
 
 	statusScanRefRange(ref)
 
-	Print("\nGit LFS objects to be committed:\n")
+	entries, errs, err := scanIndex(scanIndexAt)
+	if err != nil {
+		ExitWithError(err)
+	}
 
-	var unstagedPointers []*lfs.WrappedPointer
-	indexScanner := lfs.NewGitScanner(func(p *lfs.WrappedPointer, err error) {
-		if err != nil {
-			ExitWithError(err)
-			return
+	staged := make([]*lfs.DiffIndexEntry, 0)
+	unstaged := make([]*lfs.DiffIndexEntry, 0)
+
+L:
+	for {
+		select {
+		case entry, ok := <-entries:
+			if !ok {
+				break L
+			}
+
+			switch entry.Status {
+			case lfs.StatusModification:
+				unstaged = append(unstaged, entry)
+			default:
+				staged = append(staged, entry)
+			}
+		case err := <-errs:
+			if err != nil {
+				ExitWithError(err)
+			}
 		}
+	}
 
-		switch p.Status {
-		case "R", "C":
-			Print("\t%s -> %s (%s)", p.SrcName, p.Name, humanizeBytes(p.Size))
-		case "M":
-			unstagedPointers = append(unstagedPointers, p)
+	Print("\nGit LFS objects to be committed:\n")
+	for _, entry := range staged {
+		switch entry.Status {
+		case lfs.StatusRename, lfs.StatusCopy:
+			Print("\t%s -> %s", entry.SrcName, entry.DstName)
 		default:
-			Print("\t%s (%s)", p.Name, humanizeBytes(p.Size))
+			Print("\t%s", entry.SrcName)
 		}
-	})
-
-	if err := indexScanner.ScanIndex(scanIndexAt, nil); err != nil {
-		ExitWithError(err)
 	}
 
-	indexScanner.Close()
-
 	Print("\nGit LFS objects not staged for commit:\n")
-	for _, p := range unstagedPointers {
-		if p.Status == "M" {
-			Print("\t%s", p.Name)
-		}
+	for _, entry := range unstaged {
+		Print("\t%s", entry.SrcName)
 	}
 
 	Print("")
 }
 
+func scanIndex(ref string) (<-chan *lfs.DiffIndexEntry, <-chan error, error) {
```

Comment:
@technoweenie good call. My original intent was that this could be consumed asynchronously in case the number of `diff-index` entries was large, but since they get collected into a slice anyway, I don't think it matters.

I went ahead and made this change in 0fb059d, and I think that this PR is now ready to merge. 👍